src/Containers/precompile.jl

```diff
@@ -0,0 +1,93 @@
+    Base.precompile(Tuple{typeof(_extract_kw_args),Tuple{}})   # time: 0.010988676
+    Base.precompile(Tuple{typeof(container),Function,VectorizedProductIterator{Tuple{StepRange{Int64, Int64}, StepRange{Int64, Int64}, Base.OneTo{Int64}}},Type{DenseAxisArray}})   # time: 0.010785832
+    Base.precompile(Tuple{typeof(container),Function,VectorizedProductIterator{Tuple{Base.OneTo{Int64}, UnitRange{Int64}, UnitRange{Int64}}},Type{DenseAxisArray}})   # time: 0.010772673
+    Base.precompile(Tuple{typeof(_extract_kw_args),Tuple{Expr}})   # time: 0.010699514
+    Base.precompile(Tuple{typeof(container),Function,VectorizedProductIterator{Tuple{Vector{String}}},Type{DenseAxisArray}})   # time: 0.009808394
+    Base.precompile(Tuple{typeof(_extract_kw_args),Tuple{Symbol}})   # time: 0.009082955
+    Base.precompile(Tuple{typeof(container),Function,VectorizedProductIterator{Tuple{UnitRange{Int64}}},Type{DenseAxisArray}})   # time: 0.008933888
+    Base.precompile(Tuple{typeof(_extract_kw_args),Tuple{Symbol, Expr, Expr}})   # time: 0.007755594
+    Base.precompile(Tuple{typeof(_extract_kw_args),Tuple{Expr, Expr}})   # time: 0.007691665
+    Base.precompile(Tuple{typeof(_extract_kw_args),Tuple{Expr, Expr, Expr}})   # time: 0.006882744
+    Base.precompile(Tuple{typeof(container_code),Vector{Any},Expr,Expr,Symbol})   # time: 0.005621142
+    # TODO: isdefined(JuMP.Containers, Symbol("#37#38")) && Base.precompile(Tuple{getfield(JuMP.Containers, Symbol("#37#38")),Int64})   # time: 0.004817874
+    Base.precompile(Tuple{typeof(to_index),DenseAxisArray,Any})   # time: 0.004183273
+    Base.precompile(Tuple{typeof(vectorized_product),Base.OneTo{Int64},Vararg{Any, N} where N})   # time: 0.00327413
+    Base.precompile(Tuple{typeof(_to_index_tuple),Tuple{Int64, Vararg{Any, N} where N},Tuple{Dict{Int64, Int64}, Dict{Symbol, Int64}}})   # time: 0.003025034
+    let fbody = try __lookup_kwbody__(which(nested, (Function,Vararg{Function, N} where N,))) catch missing end
+        if !ismissing(fbody)
+            precompile(fbody, (Function,typeof(nested),Function,Vararg{Function, N} where N,))
+        end
+    end   # time: 0.002819959
+    Base.precompile(Tuple{typeof(haskey),NoDuplicateDict{_A, _B} where {_A, _B},Tuple{Int64, Int64, Int64}})   # time: 0.002722087
+    Base.precompile(Tuple{typeof(vectorized_product),Base.OneTo{Int64},Vararg{Base.OneTo{Int64}, N} where N})   # time: 0.002584661
+    Base.precompile(Tuple{typeof(_to_index_tuple),Tuple{Int64, Function},Tuple{Dict{Int64, Int64}, Dict{Symbol, Int64}}})   # time: 0.002516737
+    Base.precompile(Tuple{typeof(_to_index_tuple),Tuple{String, Int64, Int64},Tuple{Dict{String, Int64}, Dict{Int64, Int64}, Dict{Int64, Int64}}})   # time: 0.002253078
+    Base.precompile(Tuple{typeof(has_dependent_sets),Vector{Any},Vector{Any}})   # time: 0.00217896
+    # TODO: Base.precompile(Tuple{Type{DenseAxisArray},Core.Array{T, N},Any,Tuple{Vararg{Dict, N}} where N})   # time: 0.001977609
```

Comment:
@timholy I found that `SnoopCompile.parcel(tinf)` created a lot of these invalid precompile statements. Is this a bug? Expected behavior? 

Here's an example of another one: https://github.com/jump-dev/MathOptInterface.jl/pull/1243#discussion_r578863220

(See the top of the PR for how I created these files.)

Reply:
No, that's a bug. I haven't yet tried to reproduce this but this looks sufficiently detailed that I expect it to be straightforward. Thanks!!